BlazarUI/app/scripts/components/shared/ModuleModal.jsx

```diff
@@ -0,0 +1,83 @@
+import React, {Component, PropTypes} from 'react';
+import Immutable from 'immutable';
+import Modal from 'react-bootstrap/lib/Modal';
+import Button from 'react-bootstrap/lib/Button';
+import Checkbox from './Checkbox.jsx';
+import $ from 'jquery';
+
+class ModuleModal extends Component {
+
+  renderModule(moduleObject) {
+    let moduleId = moduleObject.get('id');
+    let moduleName = moduleObject.get('name');
+
+    return (
+      <div>
+        <Checkbox
+          label={moduleName}
+          name='module-checkbox'
+          value={moduleId}
+          checked={true} />
+        <br />
+      </div>
+    );
+  }
+
+  renderDownstreamToggle() {
+    return (
+      <Checkbox
+        label='Build Downstream Modules'
+        name='downstream-checkbox' />
+    );
+  }
+
+  renderModuleList() {
+    return (
+      <div className='module-modal--container'>
```

Comment:
if you're getting at using the BEM syntax (here and below), the `--` should represent a modifier, not element, so it looks like you'd want `module-modal` as the block, `module-module__container` as the element. though in this case I would just use something like `module-modal` (dropping `container`), and with the elements below, `module-modal__modules` `module-modal__downstream`, etc. Keep `--` as modifiers, eg if you were to have `modal-modal__title` as BE, a modifier would look like `modal-modal__title--big` `modal-modal__title--small` for arbitrary example.